src/libraries/Directory.Build.targets

```diff
@@ -299,4 +299,49 @@
     </PropertyGroup>
   </Target>
 
+  <PropertyGroup>
+    <BeforePack>IncludeAnalyzersInPackage;$(BeforePack)</BeforePack>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Ensure AnalyzerReference items are restored and built 
+         The target framework of Analyzers has no relationship to that of the refrencing project, 
+         so we don't apply TargetFramework filters nor do we pass in TargetFramework.
+         When BuildProjectReferences=false we make sure to set BuildReference=false to make
+         sure not to try to call GetTargetPath in the outerbuild of the analyzer project. -->
+    <ProjectReference Include="@(AnalyzerReference)" SkipGetTargetFrameworkProperties="true" UndefineProperties="TargetFramework" ReferenceOutputAssembly="false" PrivateAssets="all" BuildReference="$(BuildProjectReferences)"  />
+  </ItemGroup>
+
+  <Target Name="IncludeAnalyzersInPackage" Condition="'@(AnalyzerReference)' != ''">
+    <!-- Call a target in the analyzer project to get all the files it would normally place in a package.
+         These will be returned as items with identity pointing to the built file, and PackagePath metadata
+         set to their location in the package.  IsSymbol metadata will be set to distinguish symbols. -->
+    <MSBuild Projects="@(AnalyzerReference)"
+             Targets="GetAnalyzerPackFiles">
+      <Output TaskParameter="TargetOutputs" ItemName="_AnalyzerFiles" />
```

Comment:
We usually use the singular form instead of plural for items. I know that there are exceptions to the rule but in general we try to stick to it, i.e. `Content`, `Compile`, etc.

Reply:
Initially I intentionally used plural here to avoid confusion with item in the referenced project.  I since renamed `_AnalyzerFile` to `_AnalyzerPackFile` below, so I could drop the plural now as it's no longer needed.